File: 1.Data structure/4.Hashmap/신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    report = list(set(report))
    reporters = {}
    reportees = {}

    reportersCount = {x:0 for x in id_list}
    reporteesCount = {x:0 for x in id_list}

    for i in range(len(report)):
        temp = report[i].split(" ")
        reporters[i] = temp[0]
        reportees[i] = temp[1]
        reporteesCount[temp[1]] += 1

    for i in range(len(report)):
        if reporteesCount[reportees[i]] >= k:
            reportersCount[reporters[i]] +=1

    return list(reportersCount.values())

File: 1.Data structure/4.Hashmap/신고 결과 받기/신고 결과 받기.java
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public Integer[] solution(String[] id_list, String[] report, int k) {
        Map<String, Integer> reporter = new HashMap<>();
		Map<String, Integer> reportee = new HashMap<>();
	
		
		report = Arrays.stream(report).distinct().toArray(String[]::new);
		
		for(String s: report) {
			String[] temp = s.split(" ");
			reportee.put(temp[1], reportee.getOrDefault(temp[1], 0)+1);
		}
		
		for(String s: report) {
			String[] temp = s.split(" ");
			if (reportee.getOrDefault(temp[1], -1) >= k) {
				reporter.put(temp[0], reporter.getOrDefault(temp[0], 0)+1);
			}
		}
		
		return Arrays.stream(id_list).map(i -> reporter.getOrDefault(i,0)).toArray(Integer[]::new);
    }
}